data_and_models/pipelines/ner/transformers_vs_spacy/transformers/2_eval_pred.py

```diff
@@ -0,0 +1,97 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import json
+from functools import reduce
+from pprint import pprint
+
+import pandas as pd
+from seqeval.metrics import f1_score, accuracy_score, precision_score, recall_score, classification_report
+from seqeval.metrics import performance_measure
+from seqeval.scheme import IOB2
+
+from bluesearch.mining.eval import ner_report
```

Comment:
I am a bit confused.

We end up reimplementing part of what could be found in famous packages to reduce the number of dependencies (cf. #330, #325). But here, a new dependency is introduced (i.e. `seqeval`) while there is already an implementation for this in `mining/eval.py`.

That seems confusing...

Reply:
Agree with @pafonta . As commented above, (1) are the metrics obtained using the two approaches the same? and (2) should we consider dropping `bluesearch.mining.eval`?

Reply:
Just to give you a bit of context: the output from `transformers` is a pure classification output. There is no `post-processing` on the results (which means that a token can have the class label `I-ENT` even if the previous one is not `B-ENT` or `I-ENT`). Currently, our implementation of the metrics in `bluesearch.mining.eval` requires this consistency (for the `entity` mode). As we saw `seqeval` was the library used in the scripts of the training by `transformers` (and seems to handle the inconsistency of annotations), we decided to try it and compare with our results (in the `token` mode at least). Currently, the results are not the same as the one we obtain. This is definitely something we can investigate

> should we consider dropping bluesearch.mining.eval ?

I am not sure it is needed to use `seqeval` (especially if we go for `spaCy`). However, if we want to keep it, it is important to understand some parameters (different annotations type, how they handle the inconsistencies of label, mode strict or not) and dig into their code. Moreover, it is not sure yet if and how to compute what we call the `entity` and `token` levels.

Reply:
Opened issue #336  to transition to `seqeval` — once we make sure that it's a good idea.